### ISSUE TITLE
Excon::Error::Socket: no cipher match (OpenSSL::SSL::SSLError)

### ISSUE THREAD

**Issue by SeanRoy** [fog/fog#4008]
I am running JRuby 9.2, fog (0.7.2), fog-core (2.1.0), excon/0.62.0 . Capistrano uses fog to upload some files to S3, but I'm getting this error: Excon::Error::Socket: no cipher match (OpenSSL::SSL::SSLError).  Here's a [gist](https://gist.github.com/SeanRoy/606171353496048e75a0d775d3473e90) of the error with EXCON_DEBUG=true.  This seems to be the same problem: https://github.com/excon/excon/issues/528.

I've tried adding the suggested workaround to the initialize method in fog/core/connection.rb and that seems to get me past this, but I doubt that's the correct solution.

**Comment by geemus** [comment]
Yeah, not sure if this is a jruby issue or perhaps an openssl version issue? You might try updating openssl if everything else is up to date already, it shouldn't hurt anything either way.

That said, I believe there should be a somewhat nicer workaround at least. I think you can change defaults to get what you want, something like:

`Excon.defaults[:ciphers] = 'DEFAULT'`

I would suggest using something more explicit than default, but I'm not sure from the error which cipher(s) are failing to match. You could try excluding some portion instead of just falling back to default. In any event, if you set this, fog should use it (so you can avoid overwriting any code).

Hope that helps, certainly happy to discuss further if I can help though. Thanks!
